packages/CodingStyle/tests/Rector/ClassMethod/MakeInheritedMethodVisibilitySameAsParentRector/Fixture/skip_static_ctor.php.inc

```diff
@@ -2,21 +2,25 @@
 
 namespace Rector\CodingStyle\Tests\Rector\ClassMethod\MakeInheritedMethodVisibilitySameAsParentRector\Fixture;
 
-class SkipStaticCtor extends ParentWithPublicConstructor
+abstract class SkipStaticCtor extends ParentWithPublicConstructor
 {
     protected function __construct()
     {
+        // do something basic
     }
 
     public static function create()
     {
-        return new self();
+        return new static();
     }
 }
 
-class ParentWithPublicConstructor
+class InheritedSkipCtor extends SkipStaticCtor
 {
-    public function __construct()
+    protected function __construct()
     {
+        parent::__construct();
+
+        // do something more
     }
```

Comment:
Why this change? It refers some `ParentWithPublicConstructor` class that is now missing here

Reply:
I've tried to make a difference between the Self and Static classes. Self extends just one class (ParentWithPublicConstructor), is final, private constructor, while Static is abstract, extended by another class (InheritedSkipCtor), protected constructor.

That way it matches "real code examples" better, IMHO.

Reply:
Better add new case fixture to cover this with standalone isolated classes.
In future one of test fixture might change it would break things

Reply:
Sorry, but I don't get the issue here. I have two features (covering "return new self();" and "return new static();"). Therefore I have two fixtures/test cases files.

What more would you like to "isolate" ?

Reply:
Keep original fixture untouched + create a new file for your case.

The reason is to make debugging simpler in the future.

Reply:
Oh, I see it now. You use one class in 2 tests.
In that case it should be extracted to `Source` directory

Reply:
I'll extract it then.

Reply:
Thank you